### ISSUE TITLE
Package.json -> type: "module" -- not supported

### ISSUE THREAD

**Issue by kingmesal** [mrjono1/joi-to-typescript#231]
I recently needed to add `type:"module"` to my `package.json` and setup my `modules: "ESNext"`. This has caused joi-to-typescript to break with the following error.

TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /scripts/makeTypes.ts
    at new NodeError (node:internal/errors:372:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:56:11)
    at defaultGetFormat (node:internal/modules/esm/get_format:83:38)
    at defaultLoad (node:internal/modules/esm/load:21:20)
    at ESMLoader.load (node:internal/modules/esm/loader:407:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:326:22)
    at new ModuleJob (node:internal/modules/esm/module_job:66:26)
    at ESMLoader.#createModuleJob (node:internal/modules/esm/loader:345:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:304:34)
    at async Promise.all (index 0) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}

Now, if I add this to my tsconfig.json:
```    "ts-node": {
        "compilerOptions": {
            "module": "CommonJS"
        }
    },
```
and remove `type: "module"` from `package.json`

While I can clearly remove / readd the type module to my package.json, I was wondering if there is a build target fix that would resolve this... e.g. not commonjs module.

Any thoughts?

**Comment by mrjono1** [comment]
its sounding like I should drop commonjs module and just go esm

**Comment by luiskugel** [comment]
Hi, I just wanted to ask if there was an update on this. I would really like to use this package, but I have to use esm. So changing back to cjs is not an option for me. 

Is there any way to help you with this?

**Comment by mrjono1** [comment]
yes, if you have time to create a PR to fix this that would be awesome

**Comment by luiskugel** [comment]
Ok, I just have to get a release out the next week. When that is done in about 2 weeks, I will make the pull request. 

I have already forked and implemented it, just have to make it nice and create the pull request.

**Comment by mrjono1** [comment]
awesome thankyou, there will be many people thanking you

**Comment by Pnlvfx** [comment]
updates?

**Comment by mrjono1** [comment]
Im happy for this project to move to esm only

**Comment by throrin19** [comment]
Any news about this ? I have same problem on my project I moved to full ESM

**Comment by brenr** [comment]
Any updates on this?

**Comment by mrjono1** [comment]
I haven't had a chance to look at this I am open to PRs and new maintainers
